integration-libs/opf/base/components/opf-quick-buy/apple-pay/apple-pay.service.ts

```diff
@@ -0,0 +1,491 @@
+            ...result,
+            errors: [
+              this.updateApplePayFormWithError(
+                'No shipment methods available for this delivery address'
+              ),
+            ],
+          });
+        }
+        const newShippingMethods = modes.map((mode) => {
+          return {
+            identifier: mode.code as string,
+            label: mode.name as string,
+            amount: (mode.deliveryCost?.value as number).toFixed(2),
+            detail: mode.description ?? (mode.name as string),
+          };
+        });
+        result.newShippingMethods = newShippingMethods;
+        return result;
+      }),
+      switchMap(() => {
+        return this.cartHandlerService.getCurrentCartTotalPrice();
+      }),
+      switchMap((price: number | undefined) => {
+        if (!price) {
+          return throwError(() => new Error('Total Price not available'));
+        }
+        this.transactionDetails.total.amount = price.toString();
+        result.newTotal.amount = price.toString();
+        return of(result);
+      })
+    );
+  }
+
+  private handlePaymentMethodSelected(
+    _event: ApplePayJS.ApplePayPaymentMethodSelectedEvent
+  ): Observable<any> {
+    const result: ApplePayJS.ApplePayPaymentMethodUpdate =
+      this.updateApplePayForm({ ...this.transactionDetails.total });
+    return of(result);
+  }
+
+  private handleShippingMethodSelected(
+    _event: ApplePayJS.ApplePayShippingMethodSelectedEvent
+  ): Observable<any> {
+    const result: ApplePayJS.ApplePayShippingContactUpdate =
+      this.updateApplePayForm({ ...this.transactionDetails.total });
+
+    return this.cartHandlerService
+      .setDeliveryMode(_event.shippingMethod.identifier)
+      .pipe(
+        switchMap(() => this.cartHandlerService.getCurrentCart()),
+        take(1),
+        switchMap((cart: Cart) => {
+          if (!cart?.totalPrice?.value) {
+            return throwError(() => new Error('Total Price not available'));
+          }
+          result.newTotal.amount = cart.totalPrice.value.toString();
+          this.transactionDetails.total.amount =
+            cart.totalPrice.value.toString();
+          return of(result);
+        })
+      );
+  }
+
+  private handlePaymentAuthorized(
+    event: ApplePayJS.ApplePayPaymentAuthorizedEvent
+  ): Observable<ApplePayJS.ApplePayPaymentAuthorizationResult> {
+    const result: ApplePayJS.ApplePayPaymentAuthorizationResult = {
+      status: this.applePaySession.statusSuccess,
+    };
+    let orderSuccess: boolean;
+    return this.placeOrderAfterPayment(event.payment).pipe(
+      switchMap((success) => {
+        orderSuccess = success;
+        return this.cartHandlerService.loadCartAfterSingleProductTransaction(
+          this.transactionDetails,
+          success
+        );
+      }),
+      map(() => {
+        return orderSuccess
+          ? result
+          : { ...result, status: this.applePaySession.statusFailure };
+      }),
+      catchError((error) => {
```

Comment:
[MINOR]

We are mixing here a _static_ property `orderSuccess` with a _reactive_ stream, which is a bad practice and more difficult to reason about, comparing to a purely _reactive_ style used consistenly.

To get rid of a static property `orderSuccess`, we can simply nest the `map()` operator directly in the `.pipe()` of the stream returned by `this.cartHandlerService.loadCartAfterSingleProductTransaction`, see example:

```suggestion
    return this.placeOrderAfterPayment(event.payment).pipe(
      switchMap((orderSuccess) =>
        this.cartHandlerService
          .loadCartAfterSingleProductTransaction(
            this.transactionDetails,
            orderSuccess
          )
          .pipe(
            map(() =>
              orderSuccess
                ? result
                : { ...result, status: this.applePaySession.statusFailure }
            )
          )
      ),
```